qt/scientific_interfaces/Inelastic/Processor/DataProcessor.cpp

```diff
@@ -23,32 +23,44 @@ Mantid::Kernel::Logger g_log("DataProcessor");
 
 namespace MantidQt::CustomInterfaces {
 
-DataProcessor::DataProcessor(QObject *parent) : InelasticTab(parent), m_tabRunning(false) {
-  connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(tabExecutionComplete(bool)));
+DataProcessor::DataProcessor(QObject *parent, std::unique_ptr<MantidQt::API::IAlgorithmRunner> algorithmRunner)
+    : InelasticTab(parent), m_algorithmRunner(std::move(algorithmRunner)) {
+  m_algorithmRunner->subscribe(this);
 }
 
-DataProcessor::~DataProcessor() = default;
-
 void DataProcessor::setOutputPlotOptionsPresenter(std::unique_ptr<OutputPlotOptionsPresenter> presenter) {
   m_plotOptionsPresenter = std::move(presenter);
 }
 
+void DataProcessor::notifyBatchComplete(API::IConfiguredAlgorithm_sptr &algorithm, bool error) {
+  if (algorithm->algorithm()->name() != "SaveNexusProcessed") {
+    m_runPresenter->setRunEnabled(true);
+    runComplete(error);
+  }
+}
+
 void DataProcessor::clearOutputPlotOptionsWorkspaces() { m_plotOptionsPresenter->clearWorkspaces(); }
 
-void DataProcessor::setOutputPlotOptionsWorkspaces(std::vector<std::string> const &outputWorkspaces) {
-  m_plotOptionsPresenter->setWorkspaces(outputWorkspaces);
+API::IConfiguredAlgorithm_sptr DataProcessor::setupSaveAlgorithm(const std::string &wsName,
+                                                                 const std::string &filename) {
+  // Setup the input workspace property
+  auto saveProps = std::make_unique<Mantid::API::AlgorithmRuntimeProps>();
+  saveProps->setPropertyValue("InputWorkspace", wsName);
+  // Setup the algorithm
+  auto saveAlgo = AlgorithmManager::Instance().create("SaveNexusProcessed");
+  saveAlgo->initialize();
+
+  if (filename.empty())
+    saveAlgo->setProperty("Filename", wsName + ".nxs");
+  else
+    saveAlgo->setProperty("Filename", filename);
```

Comment:
I have tried but it doesn't save the data by using the ``saveProps``, I think it might have to do with what we discussed last week about the algorithm runtime props being sketchy with the save algorithm.

Reply:
Ah I see, we can revisit this at some point to try figure out why this doesn't work. Maybe when we move the "Save" button into the "OutputOptions" widget